src/finalsize.cpp

```diff
@@ -0,0 +1,122 @@
+#include "finalsize.h"
+
+#include <Rcpp.h>
+#include <RcppEigen.h>
+
+#include <iostream>
+
+// [[Rcpp::plugins(cpp11)]]
+// via the depends attribute we tell Rcpp to create hooks for
+// RcppEigen so that the build process will know what to do
+//
+// [[Rcpp::depends(RcppEigen)]]
+
+//' @title Calculate final epidemic size with RcppEigen backend
+//'
+//' @description This function calculates final epidemic size using SIR model
+//' for a heterogeneously mixing population.
+//'
+//' @param r0  Basic reproduction number.
+//' @param contact_matrix  Social contact matrix. Entry $mm_{ij}$ gives average
+//' number of contacts in group $i$ reported by participants in group j
+//' @param demography_vector  Demography vector. Entry $pp_{i}$ gives proportion
+//' of total population in group $i$ (model will normalise if needed)
+//' @param prop_initial_infected Proportion of all age groups that is initially
+//' infected. May be a single number, or a vector of proportions infected.
+//' If a vector, must be the same length as the demography vector, otherwise the
+//' vector will be recycled. Default value is 0.001 for all groups.
+//' @param prop_suscep  Proportion of each group susceptible. May be a single
+//' numeric value or a numeric vector of the same length as the demography
+//' vector.
+//' @param iterations Number of solver iterations
+//'
+//' @keywords epidemic model
+//' @export
+// [[Rcpp::export]]
+Eigen::VectorXd final_size_cpp(const double &r0,
+                               const Eigen::MatrixXd &contact_matrix,
+                               const Eigen::VectorXd &demography_vector,
+                               const Eigen::VectorXd &prop_initial_infected,
+                               Eigen::VectorXd prop_suscep,
+                               const int iterations = 30) {
+  if (prop_suscep.size() == 1) {
+    const double prop_suscep_ = prop_suscep[0];
+    prop_suscep.resize(demography_vector.size(), 1);
+    prop_suscep.fill(prop_suscep_);
+    Rcpp::Rcout << "prop_suscep = " << prop_suscep << "\n";
```

Comment:
Do we want a message here? We didn't have it in the R implementation

Reply:
Removed in https://github.com/epiverse-trace/finalsize/pull/36/commits/e12cf683ddba150a3d37a87581219ce38bc7cc76